samples/daal/cpp/oneccl/sources/pca_dense_online_distributed_oneccl.cpp

```diff
@@ -0,0 +1,193 @@
+/* file: pca_dense_distributed_oneccl.cpp */
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/*
+!  Content:
+!    C++ sample of principal component analysis (PCA) using the correlation
+!    method in the distributed processing mode
+!
+!******************************************************************************/
+
+/**
+ * <a name="DAAL-SAMPLE-CPP-PCA_DENSE_DISTRIBUTED"></a>
+ * \example pca_dense_distributed_oneccl.cpp
+ */
+
+#include "daal_sycl.h"
+#include "service_sycl.h"
+#include "oneapi/ccl.hpp"
+#include "mpi.h"
+#include "stdio.h"
+#include <memory>
+
+using namespace std;
+using namespace daal;
+using namespace daal::algorithms;
+
+typedef float algorithmFPType; /* Algorithm floating-point type */
+typedef services::SharedPtr<FileDataSource<CSVFeatureManager> > FileDataSourcePtr;
+
+/* PCA algorithm parameters */
+const size_t nProcs = 4;
+const size_t nVectorsInBlock = 25;
+
+/* Input data set parameters */
+const string dataFileNames[4] = { "./data/pca_normalized_1.csv", "./data/pca_normalized_2.csv", "./data/pca_normalized_3.csv",
+                                  "./data/pca_normalized_4.csv" };
+
+#define ccl_root 0
+
+int getLocalRank(ccl::communicator & comm, int size, int rank)
+{
+    /* Obtain local rank among nodes sharing the same host name */
+    char zero = static_cast<char>(0);
+    std::vector<char> name(MPI_MAX_PROCESSOR_NAME + 1, zero);
```

Comment:
I believe ccl library should provide specific macro/const variable for this. Isn't it?